### ISSUE TITLE
Error with downloading files to external drive

### ISSUE THREAD

**Issue by john3383** [ImagingDataCommons/SlicerIDCBrowser#39]
Hi all,

I'm trying to use your tool to download the large Visible Human DICOM file (233 GB) from the IDC database, but have not been able to do this with the extension. I am able to download the DICOM files into my native HDD, but my native HDD doesn't have room for this, so I attempted to redirect it to an external HDD by changing the directory in 3D Slicer Application Settings. For some reason, it will not download to my external HDD.

I'm not at all familiar with software development, and I'm not sure this is the right place to submit bugs like this, but hopefully you're either able to point me in the right direction or help me find a solution.

Best regards,

john3383

Here's a copy of the Python Console error that comes with my attempt to download files of any size from the IDC database onto my external HDD:

Reading D:/Tracheal Project/IDCLocal/archive.p

Downloading data:   0%|          | 0.00/4.39M [00:00<?, ?B/s]
Downloading data:   0%|          | 0.00/4.39M [00:00<?, ?B/s]
Downloading data:   0%|          | 0.00/4.39M [00:00<?, ?B/s]
Downloading data:   0%|          | 0.00/4.39M [00:00<?, ?B/s]
[Python] Download process failed with the following errors:
[Python] ERROR "cp s3://idc-open-data/a149f373-d72b-42a9-80fe-569801e1ce49/* D:/Tracheal Project/IDCLocal/nlm_visible_human_project/VHP-F/1.3.6.1.4.1.5962.1.2.2716.1672334394.26545/MR_1.3.6.1.4.1.5962.1.3.2716.3.1672334394.26545": expected source and destination arguments
[Python] Download process return non-zero exit code: 1
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/archive.p
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/manifest.csv
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/seriesUID.txt
Traceback (most recent call last):
  File "C:/Users/benja/AppData/Local/slicer.org/Slicer 5.7.0-2024-05-22/slicer.org/Extensions-32871/IDCBrowser/lib/Slicer-5.7/qt-scripted-modules/IDCBrowser.py", line 972, in onLoadButton
    self.addSelectedToDownloadQueue()
  File "C:/Users/benja/AppData/Local/slicer.org/Slicer 5.7.0-2024-05-22/slicer.org/Extensions-32871/IDCBrowser/lib/Slicer-5.7/qt-scripted-modules/IDCBrowser.py", line 1040, in addSelectedToDownloadQueue
    loadables = plugin.examine([fileList])
  File "C:\Users\benja\AppData\Local\slicer.org\Slicer 5.7.0-2024-05-22\lib\Slicer-5.7\qt-scripted-modules\DICOMLib\DICOMPlugin.py", line 157, in examine
    return self.examineForImport(fileList)
  File "C:/Users/benja/AppData/Local/slicer.org/Slicer 5.7.0-2024-05-22/bin/../lib/Slicer-5.7/qt-scripted-modules/DICOMScalarVolumePlugin.py", line 166, in examineForImport
    loadablesForFiles = self.examineFiles(files)
  File "C:/Users/benja/AppData/Local/slicer.org/Slicer 5.7.0-2024-05-22/bin/../lib/Slicer-5.7/qt-scripted-modules/DICOMScalarVolumePlugin.py", line 193, in examineFiles
    seriesUID = slicer.dicomDatabase.fileValue(files[0], self.tags["seriesUID"])
IndexError: tuple index out of range
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/archive.p
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/manifest.csv
[Qt] Could not read DICOM file:D:/Tracheal Project/IDCLocal/seriesUID.txt

**Comment by fedorov** [comment]
John, sorry for the trouble! In general, it is best to ask questions related to IDC in the IDC forum https://discourse.canceridc.dev/, but here is fine too (especially given that you are struggling with the specific tool).

First, please makes sure you are using the latest version of the extension. You can either download the latest preview version of 3D Slicer, or use stable 5.6.2. 

I suspect what happened is that you tried to download to disk and load the downloaded images into Slicer, and loading failed. I will try to make the code more robust, but in the meantime if you just want to download, you should click the button indicated by the red arrow in the image below.

<img width="420" alt="image" src="https://github.com/ImagingDataCommons/SlicerIDCBrowser/assets/313942/83654ddf-1c32-47db-9754-38b00dd30dee">

Also, to change the download location, you should use the destination directory selector in the module interface. You can change the default download location by clicking the button indicated by the red arrow below.

<img width="977" alt="image" src="https://github.com/ImagingDataCommons/SlicerIDCBrowser/assets/313942/dbac2149-b22a-431c-b8b1-070a9e2e2d05">

Please let me know if this addressed the problem you had!

**Comment by fedorov** [comment]
I made an improvement to more gracefully communicate to the user when a series failed to load into the scene. It should be available in the next preview release.

Also, note that there are some known problems with a number of DICOM series for the Visible Human dataset, you can see the details in this thread: https://discourse.canceridc.dev/t/loading-idc-converted-visible-human-dicom-images-into-slicer/581/10. Those issues affect a number of MR an some CT series. We plan to fix some of those issues in the next release of IDC data.

Thank you again for the report, and please let me know if this worked, or if you need further help!

**Comment by john3383** [comment]
Thank you for your prompt response. I have tried downloading a smaller CT file from the Visible Human file as you instructed without loading the images into Slicer (Pressing the stacked coins icon), but this has yielded a similar error. I do have the latest preview version of Slicer (5.7) and I downloaded the extension last week using the module extender, so I think it should be current as well. I did try it with the stable version of Slicer as well, with the same result. Here's a screenshot:
![image](https://github.com/ImagingDataCommons/SlicerIDCBrowser/assets/170750973/7138d8a0-237b-40c8-8ab6-f2cc0f516779)

Thank you!

Ben Johnson
